bgp/gobgp/plugin.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2017 Pantheon technologies s.r.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//Package gobgp contains Ligato GoBGP BGP Plugin implementation
+package gobgp
+
+import (
+	log "github.com/ligato/cn-infra/logging/logrus"
+	goBgpConfig "github.com/osrg/gobgp/config"
+	goBgpServer "github.com/osrg/gobgp/server"
+	"github.com/fgschwan/bgp-agent/bgp"
+	"github.com/fgschwan/bgp-agent/bgp/nlri"
+	"github.com/fgschwan/bgp-agent/utils/gobgp"
+	"github.com/fgschwan/bgp-agent/utils/parse"
+	"fmt"
+)
+
+type pluginImpl struct {
+	SessionConfig *goBgpConfig.Bgp
+	server        *goBgpServer.BgpServer
+	agentChannel  chan<- nlri.ReachableRoute
+	bgp.PluginCore
+	bgp.Session
+	stopWatch  chan bool
+	pluginID   string
+}
+
+//BGPPluginDebug is the public implementation of  GoBGP Ligato BGP Plugin and PluginDebug
+type BGPPluginDebug struct {
+	bgp.Plugin
+	bgp.Session
+}
+
+//New creates a GoBGP Ligato BGP Plugin implementation
+func New() BGPPluginDebug {
+	plugin := pluginImpl{}
+	return BGPPluginDebug{
+		Plugin:  bgp.Plugin{PluginCore: &plugin},
+		Session: &plugin,
+	}
+}
+
+func (plugin *pluginImpl) WaitForSessionEstablishment() error {
+	return gobgp.WaitForSessionEstablishment(plugin.server)
+}
+
+//Init defines initial status of the plugin using given config.
```

Comment:
Init uses plugin config and starts gobgp with dedicated goroutine for watching gobgp.

Reply:
True.

Reply:
Just update the go comment, please